pkg/ccl/backupccl/restore_planning.go

```diff
@@ -802,6 +802,45 @@ func resolveTargetDB(
 	return database.Name, nil
 }
 
+// maybeUpgradeTableDescsInSlice updates the passed slice of table descriptors
+// to use the newer 19.2-style foreign key representation, if they are not
+// already upgraded. This requires resolving cross-table FK references, which is
+// done by looking up all table descriptors in the slice provided.
+//
+// if skipFKsWithNoMatchingTable is set, FKs whose "other" table is missing from
+// the set provided are omitted during the upgrade, instead of causing an error
+// to be returned.
+func maybeUpgradeTableDescsInSlice(
```

Comment:
@ajwerner this part did _not_ rebase cleanly over the catalog, descpb, etc changes. I think I stubbed my way into something that compiles and passes tests, but this part should get strict scrutiny rather than a normal backport stamp